core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -312,7 +312,15 @@ class LogManager(logDirs: Seq[File],
       val logDirAbsolutePath = dir.getAbsolutePath
       var hadCleanShutdown: Boolean = false
       try {
-        val pool = Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir)
+        val pool = Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir, new ThreadFactory {
+          private val factory = Executors.defaultThreadFactory()
+          private val threadNumber = new AtomicInteger(1)
```

Comment:
Why we need this counter? It seems to me `log-recovery` is good enough.

Reply:
Only as a means of having unique names when >1 thread per log dir. You can't really infer anything from the number though, so I could remove it if you want.